p11-kit/gen-fixed-closures.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python
```

Comment:
Should we use `/usr/bin/env python` to support distributions which don't have it in `/usr/bin`, do autodetection of a Python interpreter in configure?

Should we also use `python3` in the shebang? You are relying on f-strings, [which are a Python 3.6+ feature](https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals) anyway. [PEP 394](https://peps.python.org/pep-0394/#for-python-script-publishers) says this:

> In cases where the script is expected to be executed outside virtual environments, developers will need to be aware of the following discrepancies across platforms and installation methods:
> 
> - Older Linux distributions will provide a python command that refers to Python 2, and will likely not provide a python2 command.
> - Some newer Linux distributions will provide a python command that refers to Python 3.
> - Some Linux distributions will not provide a python command at all by default, but will provide a python3 command by default.

Reply:
I was told [not](https://jmmv.dev/2016/09/env-considered-harmful.html) to use `/usr/bin/env python` at shebang if we want to enforce a specific version of Python interpreter. Anyway, the python version is checked in the top-level [meson.build](https://github.com/p11-glue/p11-kit/blob/91bec95e9ddbf9ba89dce13513e939bebd92ffce/meson.build#L438).

Reply:
That advise does apply for a typical Linux system, but it'll fail for example on macOS with MacPorts or Homebrew, since `/usr/bin/python` no longer exists on macOS Ventura (only `/usr/bin/python3` does), and that Python may be older than what other instances installed in `$PATH` are.

I'd agree with this advice if there was anything in your scripts that required modules or a specific version of Python, but from what I've seen you only use builtin modules, so any installation >= 3.6 should work.

It doesn't necessarily have to be `/usr/bin/env python3`, but at least `/usr/bin/python` will get you bug reports that the build doesn't work on modern macOS almost immediately.

Do you not still support the autoconf build system, too? In that case, does that also check the Python version?

Reply:
> Do you not still support the autoconf build system, too? In that case, does that also check the Python version?

Good point, added a check through `AM_PATH_PYTHON`.